django/core/management/commands/makemessages.py

```diff
@@ -153,16 +153,45 @@ def cleanup(self):
                 os.unlink(self.work_path)
 
 
+def build_text_file_content(lines_list=None, raw_contents=None):
+    """
+    Takes either a list of lines without EOL sequences (possibly already
+    returned by str.splitlines()) or a block of raw text that will be passed
+    through str.splitlines() to get Universal newlines treatment (has
+    precedence).
+
+    Returns the resulting block of text with normalised `\n` EOL sequences
+    ready to be written to disk using current platform native EOLs.
+    """
+    assert bool(lines_list is not None) != bool(raw_contents is not None)
```

Comment:
This is too clever. It tries to make sure only one of the two arguments is present, a logical xor operation.

Two possible alternatives:
* Simply don't check the condition, rely on the user reeading the docstring above.
* Replace it with a more readable check, e.g.:
```
assert lines_list is None or raw_contents is None
assert lines_list is not None or raw_contents is not None
```